core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -0,0 +1,515 @@
+        annotation.getReferences().add(((DDiagramElement) elementView).getTarget());
+        annotation.getDetails().put(IS_ELEMENT_TITLE_BLOCK, TRUE);
+      }
+      annotation.getReferences().addAll(annotationLines);
+      representation.getEAnnotations().add(annotation);
+    }
+  }
+
+  public void createTitleBlockLine(EObject titleBlock, EObject diagram) {
+    if (!isDiagramLevel(titleBlock)) {
+      DRepresentation representation = null;
+      if ((diagram instanceof DRepresentation)) {
+        representation = (DRepresentation) diagram;
+      }
+      if (representation != null) {
+        DAnnotation annotationLine = DescriptionFactory.eINSTANCE.createDAnnotation();
+        annotationLine.setSource("TitleBlockLine");
+        if (titleBlock instanceof DAnnotation) {
+          int numCols = getNumColumns((DAnnotation) titleBlock);
+          if (numCols > 0) {
+            ((DAnnotation) titleBlock).getReferences().add(annotationLine);
+            addColumnsToLine(annotationLine, representation, numCols, getNumLines((DAnnotation) titleBlock));
+          }
+        }
+      }
+    }
+  }
+
+  public void createTitleBlockColumn(EObject titleBlock, EObject diagram) {
+    if (!isDiagramLevel(titleBlock)) {
+      DRepresentation representation = null;
+      if ((diagram instanceof DRepresentation)) {
+        representation = (DRepresentation) diagram;
+      }
+      if (representation != null) {
+        if (titleBlock instanceof DAnnotation) {
+          int numLines = ((DAnnotation) titleBlock).getReferences().size();
+          List<EObject> lines = ((DAnnotation) titleBlock).getReferences();
+          for (int i = 0; i < numLines; i++) {
+            DAnnotation annotationCol = DescriptionFactory.eINSTANCE.createDAnnotation();
+            annotationCol.setSource("TitleBlockLineCol");
+            if (lines.get(i) instanceof DAnnotation) {
+              ((DAnnotation) (lines.get(i))).getReferences().add(annotationCol);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  private boolean isDiagramLevel(EObject titleBlock) {
+    for (EObject reference : ((DAnnotation) titleBlock).getReferences()) {
+      if (!(reference instanceof DAnnotation)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private void addColumnsToLine(DAnnotation annotationLine, DRepresentation representation, int numCols,
+      int lineNumber) {
+    List<DAnnotation> annotationCols = new ArrayList<DAnnotation>();
+    for (int j = 0; j < numCols; j++) {
+      DAnnotation annotationCol = DescriptionFactory.eINSTANCE.createDAnnotation();
+      annotationCol.setSource("TB_" + lineNumber + "_" + j);
+      annotationCols.add(annotationCol);
+      representation.getEAnnotations().add(annotationCol);
+    }
+    annotationLine.getReferences().addAll(annotationCols);
+    representation.getEAnnotations().add(annotationLine);
+  }
+
+  public int getNumLines(DAnnotation titleBlock) {

```

Comment:
It if is an element title block, it will reference also the semantic element so lines = lines = 1.

Reply:
Done.